safehttp/mux.go

```diff
@@ -71,20 +71,42 @@ type ServeMux struct {
 	mux *http.ServeMux
 }
 
-func registerHandlers(mux *http.ServeMux, handlers map[string]map[string]handlerConfig) {
-	for pattern, handlersPerMethod := range handlers {
-		handlersPerMethod := handlersPerMethod
-		mux.HandleFunc(pattern, func(w http.ResponseWriter, r *http.Request) {
-			cfg, ok := handlersPerMethod[r.Method]
-			if !ok {
-				http.Error(w, http.StatusText(http.StatusMethodNotAllowed), http.StatusMethodNotAllowed)
-				return
-			}
-			processRequest(cfg, w, r)
-		})
+func registerHandlers(mux *http.ServeMux, handlers map[string]map[string]*handlerConfig) {
+	for pattern, methodToHandlerConfigMap := range handlers {
+		httpHandler := generateHttpHandlerFunc(methodToHandlerConfigMap)
+		mux.HandleFunc(pattern, httpHandler)
+	}
+}
+
+// generateHttpHandlerFunc converts a handlerConfig to a classic http.HandlerFunc.
+// We need to have this map because http.Handle handles requests for any HTTP request method per default,
+// which we want to avoid in safehttp.
+func generateHttpHandlerFunc(methodToHandlerConfig map[string]*handlerConfig) http.HandlerFunc {
```

Comment:
"generate" is a filler that has little meaning here.

Maybe `stdHandler` is a better name here? Also, can it return a `http.Handler` instead of `http.HandlerFunc`?

Reply:
Done.